clang/lib/Sema/SemaExpr.cpp

```diff
@@ -229,6 +229,20 @@ bool Sema::DiagnoseUseOfDecl(NamedDecl *D, ArrayRef<SourceLocation> Locs,
                !checkAllowedSYCLInitializer(VD, /*CheckValueDependent =*/true))
         SYCLDiagIfDeviceCode(*Locs.begin(), diag::err_sycl_restrict)
             << Sema::KernelConstStaticVariable;
+      else if (IsConst && VD->hasGlobalStorage()) {
```

Comment:
I added IsConst since the reproducer had declared 'big' array as const. We can remove it, if we just want to check for all GV's.

Reply:
Thank you for the explanation. I will keep the check here.